lib/mayaUsd/ufe/UsdUndoDuplicateCommand.cpp

```diff
@@ -109,23 +98,23 @@ void UsdUndoDuplicateCommand::undo()
     // notification semantics, which require the object to be alive when
     // the notification is sent, we send a pre delete notification here.
     Ufe::ObjectPreDelete notification(
-        createSiblingSceneItem(fUfeSrcPath, fUsdDstPath.GetElementString()));
+        createSiblingSceneItem(_ufeSrcPath, _usdDstPath.GetElementString()));
 
 #ifdef UFE_V2_FEATURES_AVAILABLE
     Ufe::Scene::instance().notify(notification);
 #else
     Ufe::Scene::notifyObjectDelete(notification);
 #endif
 
-    fStage->RemovePrim(fUsdDstPath);
+    _srcPrim.GetStage()->RemovePrim(_usdDstPath);
 }
 
 void UsdUndoDuplicateCommand::redo()
 {
     // MAYA-92264: Pixar bug prevents redo from working.  Try again with USD
     // version 0.8.5 or later.  PPT, 28-May-2018.
```

Comment:
Well clearly redo() is working, so the comment is obsolete!

Reply:
@HamedSabri-adsk   This is not the comment I was talking about, I meant the comment on lines 82-84:
    // We use the source layer as the destination.  An alternate workflow
    // would be the edit target layer be the destination:
    // layer = self._stage.GetEditTarget().GetLayer()
The static duplicate function assumes that they should be on the same layer, but the constructor of the command assumes that it's the edit target.

Reply:
And for the redo comment...  I originally missed this comment, but I did see a similar comment in the test.  I removed that comment and reenabled the test for redo.

Reply:
Yeah, that comment can also be clean up. 

I believe both srcLayer and dstLayer that are being passed to SdfCopySpec should be the same. This is the _layer that is populated in the constructor. 

My thinking is that the duplicate always happen in one single layer. Kind of similar to how rename works.